### PR TITLE
feat(build): fluidattacks#1099 licensing builtin with reuse

### LINKED ISSUE #1099: Licensing builtin with reuse
Let's create a `testLicense` builtin with reuse.

We already have a custom job for it: https://github.com/fluidattacks/makes/tree/main/makes/license

### PR DESCRIPTION
**Pull request by rohaquinlop**
Fixes #1099 
- Create `testLicense` builtin with reuse
- Add job to dev pipeline

### CODE REVIEW

File: src/args/test-license/default.nix [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  __nixpkgs__,
+  makeScript,
+  ...
+}: {
+  name,
+  ...
+}:
+makeScript {
+  name = "test-license-for-${name}";
```
Comment (dsalaza4) [inline src/args/test-license/default.nix:10]:
```suggestion
  name = "test-license";
```

File: src/evaluator/modules/test-license/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      enable = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+      };
+    };
+  };
+  config = {
+    outputs = {
+      "/testLicense" = lib.mkIf config.testLicense.enable (testLicense {
+        name = "builtin";
```
Comment (dsalaza4) [inline src/evaluator/modules/test-license/default.nix:20]: unnecessary as it is static?

**Comment by dsalaza4** [comment]
Please add documentation to the following section: https://makes.fluidattacks.com/api/builtins/test/

File: src/evaluator/modules/test-license/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  testLicense,
+  ...
+}: {
+  config,
+  lib,
+  ...
+}: {
+  options = {
+    testLicense = {
+      enable = lib.mkOption {
```
Comment (dsalaza4) [inline src/evaluator/modules/test-license/default.nix:11]: unnecessary? maybe just an empty map? `testLicense = {};`

**Comment by dsalaza4** [comment]
Use proper commit syntax. You can run `m . /lintGitCommitMsg`.

**Comment by rohaquinlop** [comment]
> Please update your fork with upstream

Done.

**Comment by dsalaza4** [comment]
> > Please update your fork with upstream
> 
> Done.

Looks like you updated it using a merge commit instead of a rebase. Please make this PR to be only 1 commit ahead of upstream

**Comment by rohaquinlop** [comment]
> > > Please update your fork with upstream
> > 
> > 
> > Done.
> 
> Looks like you updated it using a merge commit instead of a rebase. Please make this PR to be only 1 commit ahead of upstream

Done

**Comment by dsalaza4** [comment]
Please remove https://github.com/fluidattacks/makes/tree/main/makes/license:

- [ ] Makes job
- [ ] From GitHub actions jobs

**Comment by dsalaza4** [comment]
Usea the same PR message (title and body) as in your commit.

**Comment by rohaquinlop** [comment]
> Please remove https://github.com/fluidattacks/makes/tree/main/makes/license:
> 
> * [ ]  Makes job
> * [ ]  From GitHub actions jobs

In dev and prod right?

**Comment by dsalaza4** [comment]
> 

Yes. As we're removing it, `/license` will not exist anymore. let's simply replace those github actions jobs that used it with the new `/testLicense`

**Comment by rohaquinlop** [comment]
> > 
> 
> Yes. As we're removing it, `/license` will not exist anymore. let's simply replace those github actions jobs that used it with the new `/testLicense`

I already added the new one following that it's sorted alphabetically

File: .github/workflows/prod.yml [hunk 4]
```diff
@@ -357,27 +357,6 @@ jobs:
         env:
           CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
 
-  linux_license:
```
Comment (dsalaza4) [inline .github/workflows/prod.yml:360]: Let's make `/testLicense` run on prod too (replace these ones).

**Comment by dsalaza4** [comment]
Fix jobs:

- [ ] `/formatNix`
- [ ] `formatYaml`
- [ ] `lintGitMailMap` It is failing in remote for some reason?

**Comment by rohaquinlop** [comment]
> 

Done.

**Comment by dsalaza4** [comment]
I'm not sure why but it looks like lintGitMailMap is asking for another entry when running on GitHub actions https://github.com/fluidattacks/makes/actions/runs/5259524687/jobs/9505213936?pr=1107

**Comment by rohaquinlop** [comment]
> I'm not sure why but it looks like lintGitMailMap is asking for another entry when running on GitHub actions https://github.com/fluidattacks/makes/actions/runs/5259524687/jobs/9505213936?pr=1107

Hmmm guess it's because of my GitHub name? I'm gonna change it 😅

**Comment by rohaquinlop** [comment]
@dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.

**Comment by dsalaza4** [comment]
> @dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.

I don't see any changes to the `.mailmap` file 🤔

**Comment by dsalaza4** [comment]
BTW, it runs every time you push a change, you should be able to see it here: https://github.com/fluidattacks/makes/pull/1107/checks

**Comment by rohaquinlop** [comment]
> > @dsalaza4 Can you re-run it? I already changed it, but I'm not sure if it's gonna work.
> 
> I don't see any changes to the `.mailmap` file 🤔

I mean, I changed my GitHub name, cause the name that was displaying the job was my GitHub name

**Review by dsalaza4 — APPROVED** [review]
LGTM!

Merging!

**Pull request merged**